app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js

```diff
@@ -136,6 +137,80 @@ acceptance("Admin - Watched Words", function (needs) {
     assert.strictEqual(query(".modal-body li .match").innerText, "Hello");
     assert.strictEqual(query(".modal-body li .tag").innerText, "greeting");
   });
+
+  test("emoji replacement", async function (assert) {
+    await visit("/admin/customize/watched_words/action/replace");
+    await click(".watched-word-test");
+    await fillIn(".modal-body textarea", "Hi there!");
+    assert.strictEqual(query(".modal-body li .match").innerText, "Hi");
+    assert.strictEqual(query(".modal-body li .replacement").innerText, "hello");
+  });
+});
+
+acceptance("Admin - Watched Words - Emoji Replacement", function (needs) {
+  needs.user();
+  needs.site({
+    watched_words_replace: {
+      "(?:\\W|^)(betis)(?=\\W|$)": {
+        replacement: ":poop:",
+        case_sensitive: false,
+      },
+    },
+  });
+  needs.pretender((server, helper) => {
+    server.get("/admin/customize/watched_words.json", () => {
+      return helper.response({
+        actions: ["block", "censor", "require_approval", "flag", "replace"],
+        words: [
+          {
+            id: 81,
+            word: "betis",
+            regexp: "(?:\\W|^)(betis)(?=\\W|$)",
+            replacement: ":poop:",
+            action: "replace",
+            case_sensitive: false,
+          },
+        ],
+        compiled_regular_expressions: {
+          block: null,
+          censor: null,
+          require_approval: null,
+          flag: null,
+          replace: [
+            {
+              "(?:\\W|^)(betis)(?=\\W|$)": {
+                case_sensitive: false,
+              },
+            },
+          ],
+        },
+      });
+    });
+  });
+
+  test("emoji renders successfully after replacement", async function (assert) {
+    await visit("/admin/customize/watched_words/action/replace");
+    // await this.pauseTest();
+    await visit("/");
```

Comment:
Nitpick, but the test can be shorter if we visit an existing topic and trigger a reply, we wouldn't need to set the category and title in that scenario.